NURESTConnection.js

```diff
@@ -1,6 +1,116 @@
 import NUObject from './NUObject';
 import NUInterceptor from './NUInterceptor';
 import { getLogger } from './Logger';
+import { Enum } from 'enumify';
+
+class ResponseCodeEnum extends Enum {
+    static getClassName() {
+        return 'ResponseCodeEnum';
+    }
+
+    static getEnumForCode (code) {
+        return ResponseCodeEnum.enumValues.filter( item => item.code === code);
+    }
+
+    getClassName() {
+        return ResponseCodeEnum.getClassName();
+    }
+
+    static getErrors(result) {
+        const { response } = result;
+        switch (response.status) {
+            case 400:
+            case 401:
+            case 403:
+            case 404:
+            case 405:
+            case 500:
+                const item = ResponseCodeEnum.getEnumForCode(response.status);
+                if (item && item.length) {
+                    const descriptions = [item[0].errors];
+                    return {
+                        data: {
+                            errors: [{
+                                descriptions
+                            }]
+                        }
+                    };
+                }
+                return response.statusText;
```

Comment:
caller uses errors.date (line 210). better to change this as `return { data: response.statusText }`

Reply:
may be ill discuss in person. 
if it is expected to do the login, then getErrors() itself should not be called. getErrors() should be called only if '!response.ok'.
my earlier comment was for the scenario where `!response.ok && !authFailure`, in which case control will go to line 210 where it would result in an exception

Reply:
@killanch The return at that point is mostly if things go really weird however that return should never be reached since every single code listed have errors, look at the case statement above: (400 401 403 404 405 500), there is default statement right below which is returning an object with data! Hope this helps!